enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,396 @@
+4. Controller gets the active tag list, removes requested tags from the list and populates a new list of tags.
+5. Controller replaces the tag list on the cloud resource.
+6. Controller updates the tag list to `AWSMetadata.status`.
+7. Controller updates ready condition to true in `MetadataStatus.status`.
+
+#### Handling of pre-existing tags or tags added using other tools
+
+In case of add operation, controller appends tags to existing tag list. If there is any tag pre-existing with same key name,
+there will be an error reported for the operation. While, in case of update operation, the pre-existing tag is updated. There is
+no distinction made between pre-existing tags and tags in `AWSMetadataSpec` for reconciliation. The active tag list is listed in the status.
+
+#### Classifier tag behaviour
+
+Classifier tags are used to identify cloud resources for which the controller should manage the tags. Classifier tags are mandatory and should
+be added to cloud resource. Controller does not support any wildcard characters in classifier strings.
+When multiple classifiers are used, a logical OR condition is applied for the classifiers.
+
+#### Reconciliation
+
+Reconciliation of tags is based on active tag list in `AWSMetadata.status.resourcetags`. Reconciliation of tags is ignored when ready condition is set to false at `CloudMetadata.MetadataStatus.status`. On deletion of AWSMetadataRef
+there is no reconciliation of tags. If AWSMetadataRef refers to non-existing `AWSMetadata` object , controller fails with error condition and updates `CloudMetadata.MetadataStatus.status` to false.
+
+#### No cloud provider configured
+
+When there is no cloud provider object configured, there will be no operation by default.
+
+#### Deleting classifier tag
+
+Controller does not delete, update or reconcile classifier tags. Controller refers them as read-only tags. 
+
+### Cloud provider authentication
+
+TBD
+
+### API Extensions
+
+#### Example
+
+```yaml
+apiVersion: metadata-controller.cloud.io/v1alpha1
+kind: CloudMetadata 
+metadata: 
+  name : metadata
+  namespace: metadata-controller
+spec:
+  cloudprovider:
+    awsref:
+      name: awstags
+      namespace: metadata-controller
+  classifiers:
+    "kubernetes.io/cluster/test-7lpkm-crhfx" : "owned"
+  controllerconfig:
+    limit: "POLICY_APPLY_PARTIAL"
+    syncevery: "10m"
+status:
+  type: ready
+  status: true
+  reason: "none"
+  message: "tags are applied successfully and can be referred by users"
+```
+
+```yaml
+apiVersion: metadata-controller.aws.io/v1alpha1
+kind: AWSMetadata
+metadata:
+  name: awstags
+  namespace: metadata-controller
+spec:
+  resourcetags:
+    "env": "test"
+    "centre": "eng"
+status:
+  type: applied
+  status: true
+  reason: "none"
+  message: "tags are applied successfully to cloud resources"
+  resourcetags:
+    "env": "test"
+    "centre": "eng"
+```
+### GO language structures
+```go
+type CloudMetadata struct {
```

Comment:
I think it would be useful to have example of multiple clouds. In the past where we've tried to include multiple clouds into a single project (eg MAPI) it's lead to some pretty awkward decisions API wise. It would be good to make sure to avoid those mistakes again. Having examples of how this would be expanded to a second cloud would aid in that discussion IMO